integration/spark-common/src/main/scala/org/apache/carbondata/spark/load/DataLoadProcessBuilderOnSpark.scala

```diff
@@ -127,9 +130,32 @@ object DataLoadProcessBuilderOnSpark {
       }
 
     // 4. Write
-    sc.runJob(sortRDD, (context: TaskContext, rows: Iterator[CarbonRow]) =>
+    sc.runJob(sortRDD, (context: TaskContext, rows: Iterator[CarbonRow]) => {
+      TaskContext.get.addTaskCompletionListener(_ => ThreadLocalSessionInfo.unsetAll())
+      val carbonSessionInfo: CarbonSessionInfo = {
+        var info = ThreadLocalSessionInfo.getCarbonSessionInfo
+        if (info == null || null == info.getSessionParams) {
```

Comment:
if info ！= null ，no need to generate a new object

Reply:
This piece of code is not required, so I have removed it.